docs/scos/dev/upgradability-services/new-documentation/method-of-extended-class-overridden-on-project-level.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Method of extended class is overridden one the project level
+description: Reference information for evaluator and upgrader tools.
+last_updated: Mar 23, 2022
+template: concept-topic-template
+---
+
+## Method of an extended class is overridden on the project level
+
+Factory, Dependency Provider, Repository, and Entity Manager methods belong to the private API.
+If you extend a core class and override one of its methods, minor releases can cause errors or unexpected changes in functionality.
+
+#### Using custom methods on the project level
+
+To avoid unexpected issues and achieve the same result, instead of overriding the core methods, introduce custom ones.
+
+#### Example of code that can cause upgradability errors
+
+For example, the extended class EvaluatorCategoryImageEntityManager overrides the core method CategoryImageEntityManager.
+
+```php
+namespace Pyz\Zed\Evaluator\Persistence;
+
+use Generated\Shared\Transfer\CategoryImageSetTransfer;
+use Spryker\Zed\CategoryImage\Persistence\CategoryImageEntityManager;
+
+class EvaluatorCategoryImageEntityManager extends CategoryImageEntityManager
+{
+    /**
+     * @param \Generated\Shared\Transfer\CategoryImageSetTransfer $categoryImageSetTransfer
+     *
+     * @return \Generated\Shared\Transfer\CategoryImageSetTransfer
+     */
+    public function saveCategoryImageSet(CategoryImageSetTransfer $categoryImageSetTransfer): CategoryImageSetTransfer
+    {
+        ...
+    }
+}
+```
+
+#### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------
+************************************************************************************************************************
+Evaluator\Business\Check\IsMethodOverridden\EntityManagerCheck
+Introduce a new custom method without usage of existing one. Override usage of the current method in all usage of public API.
+************************************************************************************************************************
+------------------------------------------------------------------------------------
+Pyz\Zed\EvaluatorSpryker\Persistence\EvaluatorSprykerCategoryImageEntityManager
+{"name":"saveCategoryImageSet","class":"Pyz\\Zed\\EvaluatorSpryker\\Persistence\\EvaluatorSprykerCategoryImageEntityManager"}
+{"parentClass":"Spryker\\Zed\\CategoryImage\\Persistence\\CategoryImageEntityManager"}
+************************************************************************************************************************
+```
+
+{% info_block warningBox "Dependency Provider exception" %}
+
+If you override a method and initialize a plugin, it does not break backward compatibility. For example, in `StorageRouterDependencyProvider`, `SprykerShopStorageRouterDependencyProvider` is overridden with a plugin introduced:
+
+<details>
+<summary markdown='span'>Example of an overridden method with a plugin</summary>
```

Comment:
Looks strange. Is it a new block? 
Shall it starts with ###?

Reply:
Or is it a part of the error?

Reply:
If it is part of the error - lets add this to a documentation. 

1. Investigate if it is possible to extend a functionality with plugins (link to "Plug and Play strategy")
2. If not possible - create your own method